cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -629,19 +630,28 @@ def index_about_information(cls, modulestore, course):
 
         # Broad exception handler to protect around and report problems with indexing
         try:
-            searcher.index(cls.DISCOVERY_DOCUMENT_TYPE, [course_info])
+
+            if course.catalog_visibility in (CATALOG_VISIBILITY_ABOUT, CATALOG_VISIBILITY_NONE):
+                searcher.remove(cls.DISCOVERY_DOCUMENT_TYPE, [course_info['id']])
```

Comment:
nit: kindly used double quotes for consistency.

Reply:
Updated the code so not needed anymore but good catch.